Regression in IndentationCheck 8.40 regarding the white spaces needed in a second line of a try/resource declaration

https://checkstyle.org/config_misc.html#Indentation

$ javac Test.java

successful compilation

$ Test.java

```java
import java.io.*;

class Test {
    void test() {
        try (InputStreamReader reader = 
            new InputStreamReader(System.in)) {         // violation
                // try block...    
        } catch (Exception e) {
            // catch block...
        }
    }
}
```

$ config.java

```xml
<?xml version="1.0"?>
<!DOCTYPE module PUBLIC
        "-//Checkstyle//DTD Checkstyle Configuration 1.3//EN"
        "https://checkstyle.org/dtds/configuration_1_3.dtd">
<module name="Checker">
  <module name="TreeWalker">
      <module name="Indentation" />
    </module>
  </module>
</module>
```

Actual


$ java -jar checkstyle-8.40-all.jar -c config.xml t2.java

```
[ERROR] /home/abhishhek/playground/Test.java:6:13: 
  'operator new' has incorrect indentation level 12, expected level should be 16. [Indentation]
```

Expected

No violations

-----

#9071 seems to have caused an unexpected side effect in the default behavior of IndentationCheck.

The following line used to be valid with Checkstyle 8.37: https://github.com/xwiki/xwiki-platform/blob/xwiki-platform-13.0/xwiki-platform-core/xwiki-platform-icon/xwiki-platform-icon-default/src/main/java/org/xwiki/icon/internal/DefaultIconSetManager.java#L129

and is now failing with

> 'new' has incorrect indentation level 16, expected level should be 20.

My configuration is https://github.com/xwiki/xwiki-commons/blob/xwiki-commons-13.0/xwiki-commons-tools/xwiki-commons-tool-verification-resources/src/main/resources/checkstyle.xml#L205

$ Test.java
```java
import java.io.*;

class Test {
    void test() {
        try (InputStreamReader reader = 
            new InputStreamReader(System.in)) {        
                // try block...    
        } catch (Exception e) {
            // catch block...
        }
    }
}
```

$ config.java
```bash
<?xml version="1.0"?>
<!DOCTYPE module PUBLIC
        "-//Checkstyle//DTD Checkstyle Configuration 1.3//EN"
        "https://checkstyle.org/dtds/configuration_1_3.dtd">
<module name="Checker">
  <module name="TreeWalker">
      <module name="Indentation" />
    </module>
  </module>
</module>

```
# Actual
$ java -jar checkstyle-8.40-all.jar -c config.xml t2.java 
```
[ERROR] /home/abhishhek/playground/Test.java:6:13: 'operator new' has incorrect indentation level 12, expected level should be 16. [Indentation]
```

# Expected
No violations


@tmortagne You can add these codes on your issue description to make it more clear. 
Here is the issue template for bug report: https://checkstyle.org/report_issue.html#How_to_report_a_bug.3F

--- a/.ci/travis/travis.sh
+++ b/.ci/travis/travis.sh
@@ -265,10 +265,8 @@ no-error-test-sbe)
   ;;
 
 no-error-xwiki)
-  # non-snapshot version until https://github.com/checkstyle/checkstyle/issues/9236
-  # CS_POM_VERSION=$(mvn -e -q -Dexec.executable='echo' -Dexec.args='${project.version}' \
-  #                   --non-recursive org.codehaus.mojo:exec-maven-plugin:1.3.1:exec)
-  CS_POM_VERSION=8.39
+  CS_POM_VERSION=$(mvn -e -q -Dexec.executable='echo' -Dexec.args='${project.version}' \
+                    --non-recursive org.codehaus.mojo:exec-maven-plugin:1.3.1:exec)
   echo version:$CS_POM_VERSION
   mvn -e clean install -Pno-validations
   mkdir -p .ci-temp/

--- a/src/main/java/com/puppycrawl/tools/checkstyle/checks/indentation/TryHandler.java
+++ b/src/main/java/com/puppycrawl/tools/checkstyle/checks/indentation/TryHandler.java
@@ -63,7 +63,8 @@ public class TryHandler extends BlockParentHandler {
     public IndentLevel getSuggestedChildIndent(AbstractExpressionHandler child) {
         final IndentLevel result;
         if (child instanceof CatchHandler
-            || child instanceof FinallyHandler) {
+            || child instanceof FinallyHandler
+            || child instanceof NewHandler) {
             result = getIndent();
         }
         else {

--- a/src/test/java/com/puppycrawl/tools/checkstyle/checks/indentation/IndentationCheckTest.java
+++ b/src/test/java/com/puppycrawl/tools/checkstyle/checks/indentation/IndentationCheckTest.java
@@ -2617,23 +2617,10 @@ public class IndentationCheckTest extends AbstractModuleTestSupport {
         final DefaultConfiguration checkConfig = createModuleConfig(IndentationCheck.class);
         checkConfig.addAttribute("tabWidth", "4");
         final String[] expected = {
-            "8:1: " + getCheckMessage(MSG_ERROR, "Object", 0, 12),
-            "10:1: " + getCheckMessage(MSG_ERROR, "(", 0, 12),
-            "13:1: " + getCheckMessage(MSG_CHILD_ERROR, "new", 0, 8),
-            "15:1: " + getCheckMessage(MSG_ERROR, "new lparen", 0, 8),
-        };
-        verifyWarns(checkConfig, getPath("InputIndentationNewHandler.java"), expected);
-    }
-
-    @Test
-    public void testNewHandler2() throws Exception {
-        final DefaultConfiguration checkConfig = createModuleConfig(IndentationCheck.class);
-        checkConfig.addAttribute("tabWidth", "4");
-        final String[] expected = {
-            "8:1: " + getCheckMessage(MSG_ERROR, "Object", 0, 12),
-            "10:1: " + getCheckMessage(MSG_ERROR, "(", 0, 12),
-            "13:1: " + getCheckMessage(MSG_CHILD_ERROR, "new", 0, 8),
-            "15:1: " + getCheckMessage(MSG_ERROR, "new lparen", 0, 8),
+            "10:1: " + getCheckMessage(MSG_ERROR, "Object", 0, 12),
+            "12:1: " + getCheckMessage(MSG_ERROR, "(", 0, 12),
+            "15:1: " + getCheckMessage(MSG_CHILD_ERROR, "new", 0, 8),
+            "17:1: " + getCheckMessage(MSG_ERROR, "new lparen", 0, 8),
         };
         verifyWarns(checkConfig, getPath("InputIndentationNewHandler.java"), expected);
     }

--- a/src/test/resources/com/puppycrawl/tools/checkstyle/checks/indentation/indentation/InputIndentationNewHandler.java
+++ b/src/test/resources/com/puppycrawl/tools/checkstyle/checks/indentation/indentation/InputIndentationNewHandler.java
@@ -1,5 +1,7 @@
 package com.puppycrawl.tools.checkstyle.checks.indentation.indentation; //indent:0 exp:0
 
+import java.io.InputStreamReader; //indent:0 exp:0
+
 public class InputIndentationNewHandler //indent:0 exp:0
 { //indent:0 exp:0
     public void test() { //indent:4 exp:4
@@ -13,5 +15,9 @@ Object(); //indent:0 exp:12 warn
 Integer("".indexOf("5")); //indent:0 exp:8 warn
         o = new Integer//indent:8 exp:8
 ("".indexOf("5")); //indent:0 exp:8 warn
+        try (InputStreamReader reader = //indent:8 exp:8
+            new InputStreamReader(System.in)) { //indent:12 exp:12
+        } catch (Exception e) { //indent:8 exp:8
+        } //indent:8 exp:8
     } //indent:4 exp:4
 } //indent:0 exp:0

